app/src/main/java/com/example/travellens/Filter.java

```diff
@@ -27,6 +47,48 @@ public static Filter getInstance() {
         return instance;
     }
 
+    // complex filtering algorithm
+    public List<Post> getPostsByFilteringMostRecommended(List<String> types, List<Post> currentPosts) {
+        PriorityQueue<WeightedPost> weightedPostPriorityQueue = new PriorityQueue<WeightedPost>(new Comparator<WeightedPost>() {
+            @Override
+            public int compare(WeightedPost thisPost, WeightedPost otherPost) {
+                // implement custom comparator for my weighted post
+                return Double.compare(otherPost.weight, thisPost.weight);
+            }
+        });
+
+        // iterate through each post and see how many filters it contains, calculate those weights
+        for (int post = 0; post < currentPosts.size(); post++) {
```

Comment:
This is little confusing, in this line post doesn't have post info rather the current post index.. can we update the running index to postIndex?